torch/_inductor/output_code.py

```diff
@@ -0,0 +1,179 @@
+"""
+This provides an abstract class which parametrizes over an "output code" concept
+for Inductor.  Intuitively, this represents the compiled callable which Inductor
+produces which you can call to get optimized code.  However, this callable
+has some other capabilities:
+
+- It is serializable, so you can save/load this product from disk without
+  having to do compilation again.
+
+- (When using remote cache) it is addressable, so you can save just a key
+  which you can use to load this product from remote cache later.
+
+This class is abstract because we have several different implementations of
+serialized format:
+
+- Python wrapper (the default)
+
+- AOTInductor (this produces ABI stable binaries which work across PyTorch
+  versions)
+
+"""
+
+from __future__ import annotations
+
+import dataclasses
+from typing import (
+    Any,
+    Callable,
+    Counter,
+    Dict,
+    List,
+    Optional,
+    Protocol,
+    Sequence,
+    Set,
+    Tuple,
+    TYPE_CHECKING,
+)
+from typing_extensions import TypeAlias
+
+from .runtime.autotune_cache import AutotuneCacheBundler
+
+
+if TYPE_CHECKING:
+    import torch
+    from torch._inductor import metrics
+    from torch._inductor.cudagraph_utils import BoxedDeviceIndex, CudagraphCachedInfo
+    from torch._inductor.graph import GraphLowering
+
+    from .compile_fx import _CompileFxKwargs
+    from .triton_bundler import TritonKernelArtifacts
+
+
+class OutputCode(Protocol):
+    def __call__(self, inputs: Sequence[Any]) -> Any:
+        ...
+
+
+_StrideExprStr: TypeAlias = str
+
+
+def has_frozen_params(gm: torch.fx.GraphModule) -> bool:
```

Comment:
It probably shouldn't live straight on GraphModule, because it's a very Inductor specific concept. But the real reason is I was just shoving code around lol.